nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -710,6 +718,27 @@ in
         LogsDirectoryMode = "0750";
         # Capabilities
         AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" "CAP_SYS_RESOURCE" ];
+        CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" "CAP_SYS_RESOURCE" ];
+        # Security
+        NoNewPrivileges = true;
+      } // optionalAttrs cfg.enableSandbox {
+        # Sandboxing
+        ProtectSystem = "strict";
+        ProtectHome = mkDefault true;
```

Comment:
How about moving `ProtectSystem`, `ProtectHome` and `MemoryDenyWriteExecute` to `enableSandbox` and move other options above the conditional? I think those 3 options  are the controversial ones that could break with peoples setup, where we need more feedback.

Reply:
This variant?
```
        # Security
        NoNewPrivileges = true;
        # Sandboxing
        PrivateTmp = true;
        PrivateDevices = true;
        ProtectHostname = true;
        ProtectKernelTunables = true;
        ProtectKernelModules = true;
        ProtectControlGroups = true;
        RestrictAddressFamilies = [ "AF_UNIX" "AF_INET" "AF_INET6" ];
        LockPersonality = true;
        RestrictRealtime = true;
        RestrictSUIDSGID = true;
        PrivateMounts = true;
        # System Call Filtering
        SystemCallArchitectures = "native";
      } // optionalAttrs cfg.enableSandbox {
        ProtectSystem = "strict";
        ProtectHome = mkDefault true;
        MemoryDenyWriteExecute = !(builtins.any (mod: (mod.allowMemoryWriteExecute or false)) pkgs.nginx.modules);
      };

```

Reply:
Yes. This looks good to me.

Reply:
In this variant, it will not work to disable the options `PrivateTmp` and etc.

Reply:
Right. Maybe `PrivateTmp` should be also in the `sandbox` section.

Reply:
Or leave the current variant, or rename `cfg.enableSandbox` to `cfg.enableSandboxStrict`.

Reply:
I prefer the current variant.